apps/src/templates/TeacherSectionOption.jsx

```diff
@@ -0,0 +1,75 @@
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+import color from '@cdo/apps/util/color';
+import {sectionForDropdownShape} from './teacherDashboard/shapes';
+
+export default class TeacherSectionOption extends Component {
+  static propTypes = {
+    section: sectionForDropdownShape,
+    onChange: PropTypes.func.isRequired,
+    isChecked: PropTypes.bool,
+    assignedSections: PropTypes.arrayOf(sectionForDropdownShape)
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      updatedAssignedSections: this.props.assignedSections
```

Comment:
It doesn't look like you use this? 

On a higher level, passing in `assignedSections` to each `TeacherSectionOption` is likely causing performance problems as that prop changes frequently. React will re-render this component any time the props change, so it is re-rendering all `TeacherSectionOption` components whenever one box is checked or unchecked, so it would be great to not need `assignedSections` as a prop anymore.